docs/connector-development/connector-builder-ui/tutorial.md

```diff
@@ -0,0 +1,202 @@
+      "AED": 3.6723,
+      "AFN": 86.286329,
+      "ALL": 102.489617,
+      "AMD": 391.984204,
+      // ...
+    }
+}
+```
+
+## Adding incremental reads
+
+We now have a working implementation of a connector reading the latest exchange rates for a given currency.
+In this section, we'll update the source to read historical data instead of only reading the latest exchange rates.
+
+According to the API documentation, we can read the exchange rate for a specific date range by querying the `"/exchangerates_data/{date}"` endpoint instead of `"/exchangerates_data/latest"`.
+
+To configure your connector to request every day individually, follow these steps:
+* Enable "Incremental sync" for the Rates stream
+* Set the "Cursor field" to `date` - this is the property in our records to check what date got synced last
+* Set the "Datetime format" to `%Y-%m-%d` to match the format of the date in the record returned from the API
+* Set the "Cursor granularity" to `P1D` to tell the connector the API only supports daily increments
+* Make the start time configurable by clicking the user icon in the "Start datetime" input and creating a new user input called `Start date`
+* Set the "End datetime" to `{{ today_utc() }}` to always sync up to the current day
+* In a lot of cases the start and end date are injected into the request body or request parameters. However in the case of the exchange rate API it needs to be added to the path of the request, so disable the "Inject start/end time into outgoing HTTP request" options
+* On top of the form, change the "Path URL" input to `/exchangerates_data/{{ stream_slice.start_time }}` to inject the date to fetch data for into the path of the request
+* Open the "Advanced" section and set "Step" to `P1D` to configure the connector to do one separate request per day
+* Set a start date (like `2023-03-03`) in the "Testing values" menu
+* Hit the "Test" button to trigger a new test read
+
+Now, you should see a dropdown above the records view that lets you step through the daily exchange rates along with the requests performed to fetch this data. Note that in the connector builder at most 5 slices are requested to speed up testing. During a proper sync the full time range between your configured start date and the current day will be executed.
+
+When used in a connection, the connector will make sure exchange rates for the same day are not requested multiple times - the date of the last fetched record will be stored and the next scheduled sync will pick up from where the previous one stopped.
+
+## Adding transformations
+
+Note that a warning icon should show next to the "Detected schema" tab - using the per-date endpoint instead of the latest endpoint slightly changed the shape of the records by adding a `historical` property. As we don't need this property in our destination, we can remove it using a transformation.
+
+To do so, follow these steps:
+* Enable the "Transformations" section
+* Add a new transformation
```

Comment:
Not a note for this tutorial doc, but I feel like it would be slightly more intuitive behavior if clicking the checkbox to enable Transformations caused a single empty transformation to be added automatically, rather than also requiring the user to click `+ New transformation` after enabling it. WDYT?

Reply:
That probably makes sense, could you create an issue / PR for this?